File: src/02_services/services_py_pkg/services_py_pkg/simple_srv_server.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class SimpleServiceServer(Node):
	def __init__(self):
		super().__init__('simple_service_server')

		# Create a service server
        #
        # Parameters:
        # - Service type: AddTwoInts
        # - Service name: "/add_ints"
        # - Callback function: callbackAddInts (called on every request)
		self.add_ints_service_ = self.create_service(
			AddTwoInts,
			"/add_ints",
			self.callbackAddInts
		)

		self.get_logger().info(
            "Simple Python service server node has been started"
        )

	# Callback function triggered on every received request.
    # Computes the sum of two integers and fills the response.
    #
    # @param request  Contains fields a and b
    # @param response Contains field sum
	def callbackAddInts(self, request: AddTwoInts.Request, response: AddTwoInts.Response):

		self.get_logger().info(f"Received two numbers {request.a} and {request.b}")
		response.sum = request.a + request.b

		return response


def main(args=None):
	rclpy.init(args=args)
	node = SimpleServiceServer()
	
	rclpy.spin(node)
	rclpy.shutdown()

if __name__ == '__main__':
	main()

File: src/02_services/services_cpp_pkg/src/simple_srv_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using std::placeholders::_1;
using std::placeholders::_2;

class SimpleServiceServer : public rclcpp::Node 
{
public:
	SimpleServiceServer() : Node("simple_service_server") 
	{	
		// Create a service server
		//
		// Parameters:
		// - Service type: AddTwoInts
		// - Service name: "/add_ints"
		// - Callback function: callbackAddInts (called on every request)
		add_ints_service_ = this->create_service<example_interfaces::srv::AddTwoInts>(
			"/add_ints",
			std::bind(&SimpleServiceServer::callbackAddInts, this, _1, _2)
		);

		RCLCPP_INFO(this->get_logger(), "Simple C++ service server node has been started");
	}

private:

	// Callback function triggered on every received request.
	// Computes the sum of two integers and fills the response.
	//
	// @param request  Shared pointer to the request (contains a and b)
	// @param response Shared pointer to the response (contains sum)
	void callbackAddInts(
		const example_interfaces::srv::AddTwoInts::Request::SharedPtr request,
		const example_interfaces::srv::AddTwoInts::Response::SharedPtr response
	)
	{
		RCLCPP_INFO(this->get_logger(),
			"Received two numbers %ld and %ld", request->a, request->b
		);
		response->sum = request->a + request->b;
	}

	// Service server handle for the '/add_ints' service
	rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr add_ints_service_;
};

int main(int argc, char **argv)
{
	rclcpp::init(argc, argv);
	auto node = std::make_shared<SimpleServiceServer>();
	 
	rclcpp::spin(node);
	rclcpp::shutdown();
	return 0;
} 